iosApp/widget/CommuteDepartureBoardView.swift

```diff
@@ -0,0 +1,336 @@
+        }
+        .frame(width: entry.size.width, height: entry.size.height)
+    }
+}
+
+struct CommuteStationView: View {
+    let station: DepartureBoardData.StationData
+    let entry: CommuteProvider.Entry
+    let textColor: Color
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 4) {
+            CommuteStationTitle(
+                title: station.displayName,
+                destinationStation: entry.configuration.destinationStation.toStation()?.displayName,
+                textColor: textColor,
+                maxWidth: entry.size.width
+            )
+            
+            if !station.trains.isEmpty {
+                CommuteTrainDisplay(
+                    trains: Array(station.trains.prefix(6)),
+                    entry: entry,
+                    textColor: textColor
+                )
+            } else {
+                CommuteNoTrainsView(textColor: textColor)
+            }
+        }
+    }
+}
+
+struct CommuteStationTitle: View {
+    let title: String
+    let destinationStation: String?
+    let textColor: Color
+    let maxWidth: CGFloat
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            // Station name
+            Text(
+                WidgetDataFormatter().formatHeadSign(
+                    title: title,
+                    fits: { text in
+                        // Use the widget's available width (or smaller) instead of a hardcoded value
+                        let availableWidth = maxWidth
+                        let textWidth = measureTextWidth(
+                            maxSize: CGSize(width: availableWidth, height: 50),
+                            text: text,
+                            font: UIFont.arimaStyleMediumItalic(size: 14)
+                        )
+                        return (textWidth <= availableWidth).toKotlinBoolean()
+                    }
+                )
+            )
+            .font(Font.arimaStyleMedium(size: 14))
+            .foregroundColor(textColor)
+            .italic()
+            
+            // Destination station
+            if let destination = destinationStation {
+                HStack(spacing: 2) {
+                    Image(systemName: "arrow.right")
+                        .font(.system(size: 12))
+                        .foregroundColor(textColor)
+                    Text(destination)
+                        .font(Font.arimaStyle(size: 14))
+                        .italic()
+                        .foregroundColor(textColor)
+                }
+            }
+        }
+        .padding(.leading, 4)
+        .padding(.top, 2)
+    }
+}
+
+struct CommuteTrainDisplay: View {
+    let trains: [DepartureBoardData.TrainData]
+    let entry: CommuteProvider.Entry
+    let textColor: Color
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: -2) {
+            // First train - prominent display
+            let firstTrain = trains.first!
```

Comment:
nit: I would explicitly take a non-nil first train here to make this more null-safe